airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -494,6 +495,7 @@ def load_file(
             uploaded to the S3 bucket.
         :type acl_policy: str
         """
+        filename = filename.as_posix() if isinstance(filename, Path) else filename
```

Comment:
What is the semantic meaning of `filename` here, a file on the local filesystem, or a path in the S3 bucket? If it’s the latter, this should be `Union[str, PurePosixPath]` instead, and just use `str()` for coercion.

Reply:
thanks @uranusjr 

> what is the semantic meaning of filename here, a file on the local filesystem, or a path in the S3 bucket? 

it means path to local file

> If it’s the latter, this should be Union[str, PurePosixPath] instead, and just use str() for coercion. 

did you perhaps mean former?

Reply:
No, I meant the latter. `Path` is for actual paths in the local filesystem, and shouldn’t be used to represent a path on S3. That should use `PurePosixPath`, which is basically a wrapper to a list of path components that will be joined with `/` when stringified. Your current implementation is the opposite of what I had in mind; since `filename` here should point to a local file, it should use `Path`.

Reply:
Wellp, that's how I had it in the first place

I've switched it back to `Path` now